### PR TITLE
Support if expression

### PR DESCRIPTION
**Pull request by yoshihiro503**
Support if expression for #54

### CODE REVIEW

File: lib/abstract_format.ml [hunk 1]
```diff
@@ -65,6 +65,7 @@ and expr_t =
   | ExprFun of {line: line_t; name: string option; clauses: clause_t list}
   | ExprLocalCall of {line: line_t; function_expr: expr_t; args: expr_t list}
   | ExprRemoteCall of {line: line_t;  line_remote: line_t; module_expr: expr_t; function_expr: expr_t; args: expr_t list}
+  | ExprIf of {line: line_t; if_clauses: clause_t list}
```
Comment (amutake) [inline lib/abstract_format.ml]: Could you change `if_clauses` to `clauses` according to `ExprCase` and `ExprFun`?

File: lib/abstract_format.ml [hunk 2]
```diff
@@ -688,6 +696,18 @@ and cls_of_sf ?(in_function=false) sf : (clause_t, err_t) Result.t =
      let%bind body = sf_body |> expr_of_sf |> track ~loc:[%here] in
      ClsCase {line; pattern; guard_sequence = Some guard_sequence; body} |> return
 
+  (* if clause Gs -> B *)
+  | Sf.Tuple (5, [
+                 Sf.Atom "clause";
+                 Sf.Integer line;
+                 Sf.List [];
```
Comment (amutake) [inline lib/abstract_format.ml]: I have confirmed that the 3rd element of the other non-function clauses isn't empty list. So it's OK.

File: lib/abstract_format.ml [hunk 3]
```diff
@@ -688,6 +696,18 @@ and cls_of_sf ?(in_function=false) sf : (clause_t, err_t) Result.t =
...
+                 sf_guard_sequence;
```
Comment (amutake) [inline lib/abstract_format.ml]: memo: `if clause` has non-empty guard sequence.

File: lib/abstract_format.ml (hunk 1 above)
Comment (yoshihiro503) [reply lib/abstract_format.ml]: Fixed it.

File: lib/abstract_format.ml (hunk 3 above)
Comment (yoshihiro503) [reply lib/abstract_format.ml]: Added a comment.
https://github.com/yutopp/obeam/pull/72/files#diff-38aa0e25050364494cd836c9c5f3c0d4R92

**Pull request merged**